modules/gateway/gateway.go

```diff
@@ -58,7 +59,13 @@ func (g *Gateway) Close() error {
 
 	// save the latest gateway state
 	id := g.mu.RLock()
-	if err := g.saveSync(); err != nil {
+	var err error
+	if build.Release == "testing" {
+		err = g.save()
+	} else {
+		err = g.saveSync()
+	}
```

Comment:
Oh. I still don't think you should be doing this at this level. If you don't think we should be syncing during testing, let's handle it in the persist package and not here.